pelicun/model/loss_model.py

```diff
@@ -1734,23 +1746,52 @@ def save_sample(
 
         return res
 
-    def load_sample(self, filepath: str | pd.DataFrame) -> None:
+    def load_sample(self, filepath: str | pd.DataFrame) -> dict[str, str]:
         """
         Load damage sample data.
 
+        Parameters
+        ----------
+        filepath: str
+          Path to an existing sample stored in a file, or dataframe
+          containing the existing sample.
+
+        Returns
+        -------
+        dict[str, str]
+          Dictionary mapping each decision variable to its assigned
+          unit.
+
         """
+        names = ['dv', 'loss', 'dmg', 'ds', 'loc', 'dir', 'uid']
         self.log.div()
         self.log.msg('Loading loss sample...')
 
-        sample = file_io.load_data(
-            filepath, self._asmnt.unit_conversion_factors, log=self._asmnt.log
+        sample, units = file_io.load_data(
+            filepath,
+            self._asmnt.unit_conversion_factors,
+            log=self._asmnt.log,
+            return_units=True,
+        )
+        units.index.names = names
```

Comment:
commit: https://github.com/NHERI-SimCenter/pelicun/pull/51/commits/669a917420475eadf18ba7ac9de4a10858b8b007

Similarly to my suggestion at another location, I would not force the 'uid' level here either. A simple check of the number of levels in the index would allow users to have that as an optional one.

Reply:
859c41ac